### ISSUE TITLE
Feature request:Make Envelope mutable and pooled?

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#26022]
I think the `Envelope` is nearly a totally internal thing and could be pooled and recyclable, which will help reduce the system's gc pressure, it does help.

https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/Recycler.java

**Comment by hepin1989** [comment]
another kind of attempt:
https://github.com/classgraph/classgraph/commit/42c94905349f4d41635eabe100552210570eeed2

**Comment by lukehutch** [comment]
Here's a direct link to the ClassGraph recycler... feel free to use it out of the box if you want.

https://github.com/classgraph/classgraph/tree/master/src/main/java/nonapi/io/github/classgraph/recycler

**Comment by hepin1989** [comment]
@lukehutch thanks for the link, this would be a big change to the fundamental, so I will first try with some experiment with jmh benchmark so does gc when I have time :)

**Comment by lukehutch** [comment]
Good idea to benchmark first :-)  The performance hotspots are often not where you think they are!
